### PR TITLE
feat: add K8s().Watch() reconnect on failure logic

### PR DESCRIPTION
**Pull request by jeff-mccoy**

### CODE REVIEW

File: src/fluent/watch.ts [hunk 1]
```diff
@@ -31,64 +73,134 @@ export async function ExecWatch<T extends GenericClass>(
...
+  opts.timeout = 15 * 1000;
+
+  // Enable keep alive
+  (opts.agent as unknown as { keepAlive: boolean }).keepAlive = true;
+
+  // Track the number of retries
+  let retryCount = 0;
+
+  // Set the maximum number of retries to 5 if not specified
+  watchCfg.retryMax ??= 5;
```
Comment (Noxsios) [inline src/fluent/watch.ts:86]: nullish coalescing assignment??

File: src/fluent/watch.ts [hunk 2]
```diff
@@ -31,64 +73,134 @@ export async function ExecWatch<T extends GenericClass>(
...
+    // On unhandled errors, retry the watch
+    async function reload(e: Error) {
+      // If there are more attempts, retry the watch
+      if (watchCfg.retryMax! > retryCount) {
+        retryCount++;
+
+        watchCfg.logFn?.(`retrying watch ${retryCount}/${watchCfg.retryMax}`);
+
+        // Sleep for the specified delay or 5 seconds
+        await new Promise(r => setTimeout(r, (watchCfg.retryDelaySec ?? 5) * 1000));
```
Comment (Noxsios) [inline src/fluent/watch.ts:190]: Do you want to assign the default value in the same way retryMax is so you are not doing inline assignments?

**Review by Noxsios — COMMENTED** [review]
looks cool

File: src/fluent/watch.ts (hunk 2 above)
Comment (jeff-mccoy) [reply src/fluent/watch.ts:190]: Good point, I'll update that

**Pull request merged**